### ISSUE TITLE
create a pre-commit checker that verifies doc links

### ISSUE THREAD

**Issue by marekbiskup** [GoogleCloudPlatform/kubernetes#10985]
Broken links look very bad - suggesting that docs are obsolete and nobody cares about them. Or that entire project is obsolete. I keep on finding broken links in our docs over and over again.
(like here: 
We should have a pre-commit checker that would fail if any doc link is invalid.

For the first step, let's check project-relative links - seems to be easier and maybe there are tools for that.

**Comment by erictune** [comment]
I just added some code that will help with this.
The file is called "cmd/mungedocs/mungedocs.go".
Basically, you just need to add some code around line 60 that says:
```go
	after2, err := checkLinks(after, path)
	if err != nil {
		return err
	}
```
and implement a `checkLinks(markdownDocument []byte, anchorFilePath string) error` function.
That function just needs to make sure that for each link in `markdownDocument`, that the file `dirname(anchorFilePath) + "/" + link` exists.

**Comment by erictune** [comment]
Will be fixed when #11038 merges
